### PR TITLE
Fix `qnode_spectrum` with `interface="auto"`

### LINKED ISSUE #6593: [BUG] Documentation examples on `pennylane.fourier.qnode_spectrum` no longer work
### Expected behavior

I expect to be able to run all the examples in the `pennylane.fourier.qnode_spectrum` https://docs.pennylane.ai/en/stable/code/api/pennylane.fourier.qnode_spectrum.html examples.

### Actual behavior

Examples in the documentation should be fully executable, and not provide any errors.

### Additional information

_No response_

### Source code

```shell
n_qubits = 3
dev = qml.device("default.qubit", wires=n_qubits)

@qml.qnode(dev)
def circuit(x, y, z, w):
    for i in range(n_qubits):
        qml.RX(0.5*x[i], wires=i)
        qml.Rot(w[0,i,0], w[0,i,1], w[0,i,2], wires=i)
        qml.RY(2.3*y[i], wires=i)
        qml.Rot(w[1,i,0], w[1,i,1], w[1,i,2], wires=i)
        qml.RX(z, wires=i)
    return qml.expval(qml.Z(0))

x = np.array([1., 2., 3.])
y = np.array([0.1, 0.3, 0.5])
z = -1.8
w = np.random.random((2, n_qubits, 3))
print(qml.draw(circuit)(x, y, z, w))

res = qml.fourier.qnode_spectrum(circuit, argnum=[0, 1, 2])(x, y, z, w)
for inp, freqs in res.items():
    print(f"{inp}: {freqs}")
```

### Tracebacks

```shell
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-4-a66971ae3a6a> in <cell line: 24>()
     22 print(qml.draw(circuit)(x, y, z, w))
     23 
---> 24 res = qml.fourier.qnode_spectrum(circuit, argnum=[0, 1, 2])(x, y, z, w)
     25 for inp, freqs in res.items():
     26     print(f"{inp}: {freqs}")

6 frames
/usr/local/lib/python3.10/dist-packages/numpy/core/numeric.py in within_tol(x, y, atol, rtol)
   2330     def within_tol(x, y, atol, rtol):
   2331         with errstate(invalid='ignore'), _no_nep50_warning():
-> 2332             return less_equal(abs(x-y), atol + rtol * abs(y))
   2333 
   2334     x = asanyarray(a)

ValueError: operands could not be broadcast together with shapes (27,3) (3,3)
```

### System information

```shell
Name: PennyLane
Version: 0.39.0
Summary: PennyLane is a cross-platform Python library for quantum computing, quantum machine learning, and quantum chemistry. Train a quantum computer the same way as a neural network.
Home-page: https://github.com/PennyLaneAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /usr/local/lib/python3.10/dist-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, packaging, pennylane-lightning, requests, rustworkx, scipy, toml, typing-extensions
Required-by: PennyLane_Lightning

Platform info:           Linux-6.1.85+-x86_64-with-glibc2.35
Python version:          3.10.12
Numpy version:           1.26.4
Scipy version:           1.13.1
Installed devices:
- lightning.qubit (PennyLane_Lightning-0.39.0)
- default.clifford (PennyLane-0.39.0)
- default.gaussian (PennyLane-0.39.0)
- default.mixed (PennyLane-0.39.0)
- default.qubit (PennyLane-0.39.0)
- default.qutrit (PennyLane-0.39.0)
- default.qutrit.mixed (PennyLane-0.39.0)
- default.tensor (PennyLane-0.39.0)
- null.qubit (PennyLane-0.39.0)
- reference.qubit (PennyLane-0.39.0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

### PR DESCRIPTION
**Pull request by dwierichs**
**Context:**
An example from the docstring of `qnode_spectrum` is failing because the detection of trainable arguments fails when using Autograd via vanilla numpy inputs and `interface="auto"` (default).

Note that this problem is caused by a bug in `qml.math.is_independent`, but so far only surfaces in the specific context of `qnode_spectrum`, which uses this function to check a Jacobian to be constant.

**Description of the Change:**
Add a warning to the docs that pure numpy parameters are not supported.
Raise an error if pure numpy parameters are present.
Smaller updates relating to code quality/linting.

**Benefits:**
Fixes #6593 

**Possible Drawbacks:**
N/A

**Related GitHub Issues:**
Closes #6593 
[sc-78453]

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -57,6 +57,10 @@
 
 <h3>Improvements 🛠</h3>
 
+* Raises a comprehensive error when using `qml.fourier.qnode_spectrum` with standard numpy
```
Comment (dwierichs) [inline doc/releases/changelog-dev.md:60]: Not sure we need two changelog entries?

File: pennylane/fourier/circuit_spectrum.py [hunk 2]
```diff
@@ -122,9 +122,9 @@ def circuit(x, w):
         res = qml.fourier.circuit_spectrum(circuit)(x, w)
 
     >>> print(qml.draw(circuit)(x, w))
-    0: ──RX(1.00)──Rot(0.53,0.70,0.90)──RX(1.00)──Rot(0.81,0.38,0.43)──RZ(1.00)─┤  <Z>
-    1: ──RX(2.00)──Rot(0.56,0.61,0.96)──RX(2.00)──Rot(0.32,0.49,0.77)───────────┤
-    2: ──RX(3.00)──Rot(0.11,0.63,0.31)──RX(3.00)──Rot(0.52,0.46,0.83)───────────┤
+    0: ──RX(1.00,"x0")──Rot(0.03,0.03,0.37)──RX(1.00,"x0")──Rot(0.35,0.89,0.29)──RZ(1.00,"x0")─┤  <Z>
+    1: ──RX(2.00,"x1")──Rot(0.70,0.12,0.60)──RX(2.00,"x1")──Rot(0.04,0.03,0.88)────────────────┤
+    2: ──RX(3.00,"x2")──Rot(0.65,0.87,0.05)──RX(3.00,"x2")──Rot(0.37,0.53,0.02)────────────────┤
```
Comment (dwierichs) [inline pennylane/fourier/circuit_spectrum.py:127]: Unrelated change regarding operation `id`s

File: pennylane/fourier/qnode_spectrum.py [hunk 3]
```diff
@@ -363,7 +370,7 @@ def circuit(x, y, z):
         As we can see, the preprocessing in the QNode is not included in the simple spectrum.
         In contrast, the output of ``qnode_spectrum`` is:
 
-        >>> adv_spec = qml.fourier.qnode_spectrum(circuit, encoding_args={"y", "z"})
+        >>> adv_spec = qml.fourier.qnode_spectrum(circuit, encoding_args={"y", "z"})(x, y, z)
```
Comment (dwierichs) [inline pennylane/fourier/qnode_spectrum.py:373]: The args were missing here.

File: pennylane/fourier/qnode_spectrum.py [hunk 4]
```diff
@@ -224,6 +224,12 @@ def qnode_spectrum(qnode, encoding_args=None, argnum=None, decimals=8, validatio
         as long as this processing is *linear*. In particular, constant
         prefactors for the encoding arguments are allowed.
 
+    .. warning::
```
Comment (dwierichs) [inline pennylane/fourier/qnode_spectrum.py:227]: New warning not to use pure numpy arguments.

File: pennylane/fourier/qnode_spectrum.py [hunk 5]
```diff
@@ -246,10 +252,11 @@ def circuit(x, y, z, w):
 
     This circuit looks as follows:
 
-    >>> x = np.array([1., 2., 3.])
-    >>> y = np.array([0.1, 0.3, 0.5])
-    >>> z = -1.8
-    >>> w = np.random.random((2, n_qubits, 3))
+    >>> from pennylane import numpy as pnp
```
Comment (dwierichs) [inline pennylane/fourier/qnode_spectrum.py:255]: use Autograd.

File: pennylane/math/is_independent.py [hunk 6]
```diff
@@ -110,7 +111,8 @@ def _jax_is_indep_analytic(func, *args, **kwargs):
     """
     import jax
 
-    mapped_func = lambda *_args: func(*_args, **kwargs)
+    mapped_func = partial(func, **kwargs)
```
Comment (dwierichs) [inline pennylane/math/is_independent.py:114]: Get rid of unnecessary lambda

File: pennylane/math/is_independent.py [hunk 7]
```diff
@@ -182,7 +184,10 @@ def _get_random_args(args, interface, num, seed, bounds):
         tf.random.set_seed(seed)
         rnd_args = []
         for _ in range(num):
-            _args = (tf.random.uniform(tf.shape(_arg)) * width + bounds[0] for _arg in args)
+            _args = (
+                tf.random.uniform(tf.shape(_arg), dtype=_arg.dtype) * width + bounds[0]
```
Comment (dwierichs) [inline pennylane/math/is_independent.py:188]: Pylint (or Codefactor?) was complaining about the `dtype` not being specified.

**Review by albi3ro — APPROVED** [review]
👍

**Review by JerryChen97 — APPROVED** [review]
